src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,203 @@
+        registrationService.register(expected);
+        User actual = storageDao.get(expected.getLogin());
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void register_alreadyExistedUser_notOk() {
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(INITIAL_USER_IN_STORAGE);
+        });
+        int actualSizeOfStorage = Storage.people.size();
+        assertEquals(INITIAL_SIZE_OF_STORAGE, actualSizeOfStorage);
+    }
+
+    @Test
+    void register_nullUser_notOkay() {
+        User nullUser = null;
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(nullUser);
+        });
+        int actualSizeOfStorage = Storage.people.size();
+        assertEquals(INITIAL_SIZE_OF_STORAGE, actualSizeOfStorage);
+    }
+
+    @Test
+    void register_userWithValidLogin_ok() {
+        User expected = new User(VALID_LOGIN_EXAMPLE, VALID_PASSWORD_EXAMPLE, MIN_VALID_AGE);
+        registrationService.register(expected);
+        User actual = storageDao.get(expected.getLogin());
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void register_userWithInvalidLogin_notOk() {
+        User fiveCharacterLoginUser = new User(INVALID_LOGIN_EXAMPLE1,
+                VALID_PASSWORD_EXAMPLE,
+                MIN_VALID_AGE);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(fiveCharacterLoginUser);
+        });
+
+        User fourCharacterLoginUser = new User(INVALID_LOGIN_EXAMPLE2,
+                VALID_PASSWORD_EXAMPLE,
+                MIN_VALID_AGE);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(fourCharacterLoginUser);
+        });
+
+        User oneCharacterLoginUser = new User(INVALID_LOGIN_EXAMPLE3,
+                VALID_PASSWORD_EXAMPLE,
+                MIN_VALID_AGE);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(oneCharacterLoginUser);
+        });
+
+        int actualSizeOfStorage = Storage.people.size();
+        assertEquals(INITIAL_SIZE_OF_STORAGE, actualSizeOfStorage);
+    }
+
+    @Test
+    void register_userWithNullLogin_notOk() {
+        User nullLoginUser = new User(null, VALID_PASSWORD_EXAMPLE, MIN_VALID_AGE);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(nullLoginUser);
+        });
+        int actualSizeOfStorage = Storage.people.size();
+        assertEquals(INITIAL_SIZE_OF_STORAGE, actualSizeOfStorage);
+    }
+
+    @Test
+    void register_userWithValidPassword_ok() {
+        User expected = new User(VALID_LOGIN_EXAMPLE, VALID_PASSWORD_EXAMPLE, MIN_VALID_AGE);
+        registrationService.register(expected);
+        User actual = storageDao.get(expected.getLogin());
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void register_userWithInvalidPassword_notOk() {
```

Comment:
to much asserts for test, to check invalid password one wrong user will be enough(check other your test as well)